app/controllers/posts_controller.rb

```diff
@@ -6,5 +6,24 @@ def index
 
   def show
     @post = Post.find(params[:id])
+    @author = User.find(params[:user_id])
+    @comments = Comment.where(post_id: params[:id])
+    @comment = @post.comments.new
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :text))
```

Comment:
- [ OPTIONAL ] I think it would be great if you make the parameters private in a method to make them a strong parameter you can check this [lesson](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides) to learn more about it ✨🔥